### PR TITLE
COR-525, COR-527: ContentItemFieldType Association And Selection Implementation

### PR DESCRIPTION
**Pull request by toastercup**
This PR fully implements `ContentItemFieldType` selection. Its implementation is loosely based on https://github.com/cbdr/cortex/issues/423. Things to note:
* The Cell logic is the "good" implementation - it's almost entirely generic, sans a couple trivial hardcoded bits. Even though this neat generic implementation will never be used, we can follow along when we're creating the React implementation - pretend this is a "reference" implementation that we can study later on.
* The JS bits are the "bad" parts. They're mostly hardcoded, given our current infrastructure - the server knows too much, and the client knows too little. It works, however, and its implementation is irrelevant to the platform's success.

### CODE REVIEW

File: app/assets/javascripts/ckeditor/plugins/cortex_media_insert/plugin.js [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
+(function (global) {
   'use strict';
 
   global.CKEDITOR.plugins.add('cortex_media_insert', {
     icons: 'media',
     init: function (editor) {
       editor.addCommand('insertMedia', {
         exec: function (editor) {
-          window.MODALS.featured.open();
+          window.MODALS.wysiwyg.open();
```
Comment (arelia) [inline app/assets/javascripts/ckeditor/plugins/cortex_media_insert/plugin.js:9]: why rename this line and line 16 from 'featured' or 'media' to the more general 'wysiwyg'?

File: app/assets/javascripts/cortex-field_types-core/cells/content_item.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  $('.content-item-button').append(
+    '<div class="content-item-button__selection">' +
+    '<img src="' + thumb_url + '" height="50px">' +
+    '<div class="content-item-button__selection__text">' +
+    'Selected Media: ' +
+    title +
+    '</div></div>'
+  );
+
+  window.MODALS.featured.close();
```
Comment (arelia) [inline app/assets/javascripts/cortex-field_types-core/cells/content_item.js:18]: Should this method match the one above (https://github.com/cortex-cms/cortex-plugins-core/pull/32/files#diff-f5165bb3219f906c9714100622beb28eR9) i.e. `window.MODALS.wysiwyg.close();`

File: app/assets/javascripts/ckeditor/plugins/cortex_media_insert/plugin.js (hunk 1 above)
Comment (MKwenhua) [reply app/assets/javascripts/ckeditor/plugins/cortex_media_insert/plugin.js:9]: @arelia of the tree popups featured-tile-wysiwyg this deals with images/assets to be inserted into the wysiwyg. It was a point I made in a past PR so it's good to see the fix.

File: app/assets/javascripts/cortex-field_types-core/cells/content_item.js (hunk 2 above)
Comment (MKwenhua) [reply app/assets/javascripts/cortex-field_types-core/cells/content_item.js:18]: 👍

**Review by MKwenhua — APPROVED** [review]
Looks good 👍

File: app/assets/javascripts/ckeditor/plugins/cortex_media_insert/plugin.js (hunk 1 above)
Comment (toastercup) [reply app/assets/javascripts/ckeditor/plugins/cortex_media_insert/plugin.js:9]: Featured, Tile and WYSIWYG are all 3 separate, temporarily hardcoded popups in an Employer Blog CRUD page. This plugin was opening the wrong popup (featured) for CKEditor, when it should have been opening the WYSIWYG popup - hence the change here. The featured/wysiwyg/tile objects in this chain are dynamically derived from `data` attributes on each of our 3 popup HTML elements.

File: app/assets/javascripts/cortex-field_types-core/cells/content_item.js (hunk 2 above)
Comment (toastercup) [reply app/assets/javascripts/cortex-field_types-core/cells/content_item.js:18]: See above ^

File: app/cells/plugins/core/content_item/popup.haml [hunk 3]
```diff
@@ -3,14 +3,17 @@
 
 = render_label
 %br
-%button.content_item_button.text-center.popup--open
-  .button_content
+%button.content-item-button.text-center.popup--open
```
Comment (arelia) [inline app/cells/plugins/core/content_item/popup.haml]: since you've applied centering to `.content-item-button` you can remove `.text-center`

Comment (toastercup) [reply app/cells/plugins/core/content_item/popup.haml]: @arelia Done & done! Thanks for pointing that out. Please re-review.

**Review by arelia — APPROVED** [review]

**Pull request merged**